File: src/org/elixir_lang/debugger/node/event/Listener.java
package org.elixir_lang.debugger.node.event;

import com.ericsson.otp.erlang.OtpErlangPid;
import org.elixir_lang.debugger.node.ProcessSnapshot;

import java.util.List;

public interface Listener {
  void debuggerStarted();
  void failedToInterpretModules(String nodeName, List<String> modules);
  void failedToDebugRemoteNode(String nodeName, String error);
  void unknownMessage(String messageText);
  void failedToSetBreakpoint(String module, String file, int line, String errorMessage);
  void breakpointIsSet(String module, String file, int line);
  void breakpointReached(OtpErlangPid pid, List<ProcessSnapshot> snapshots);
  void debuggerStopped();
}

File: src/org/elixir_lang/debugger/node/event/Listener.kt
package org.elixir_lang.debugger.node.event

import com.ericsson.otp.erlang.OtpErlangObject
import com.ericsson.otp.erlang.OtpErlangPid
import org.elixir_lang.debugger.InterpretedModule
import org.elixir_lang.debugger.node.ProcessSnapshot

interface Listener {
    fun breakpointIsSet(module: String, file: String, line: Int)
    fun breakpointReached(pid: OtpErlangPid, snapshots: List<ProcessSnapshot>)
    fun debuggerStarted()
    fun debuggerStopped()
    fun failedToDebugRemoteNode(nodeName: String, error: OtpErlangObject)
    fun failedToInterpretModules(nodeName: String, errorReasonByModule: Map<String, OtpErlangObject>)
    fun failedToSetBreakpoint(module: String, file: String, line: Int, errorMessage: OtpErlangObject)
    fun interpretedModules(interpretedModuleList: List<InterpretedModule>)
    fun unknownMessage(messageText: String)
}
